tests/Integration/Behaviour/Features/Context/Domain/ApiAccessManagementFeatureContext.php

```diff
@@ -0,0 +1,172 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command\AddApiAccessCommand;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command\EditApiAccessCommand;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Exception\ApiAccessConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Query\GetApiAccessForEditing;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\ValueObject\ApiAccessId;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Util\DataComparator;
+use Tests\Integration\Behaviour\Features\Context\Util\DataTransfer;
+use Tests\Resources\Resetter\ApiAccessResetter;
+
+class ApiAccessManagementFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When /^I create an api access "(.+)" with following properties:$/
+     */
+    public function createApiAccessUsingCommand($apiAccessReference, TableNode $table)
+    {
+        $data = $this->fixDataType($table->getRowsHash());
+        $commandBus = $this->getCommandBus();
+
+        $command = new AddApiAccessCommand(
+          $data['clientName'],
+          $data['apiClientId'],
+          $data['enabled'],
+          $data['description']
+        );
+
+        try {
+            /** @var ApiAccessId $id */
+            $id = $commandBus->handle($command);
+
+            $this->getSharedStorage()->set($apiAccessReference, $id->getValue());
+        } catch (ApiAccessConstraintException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then /^api access "(.+)" should have the following properties:$/
+     */
+    public function assertQueryApiAccessProperties($apiAccessReference, TableNode $table)
+    {
+        $expectedData = $table->getRowsHash();
+        $this->getSharedStorage()->exists($apiAccessReference);
+
+        $queryBus = $this->getQueryBus();
+        $result = $queryBus->handle(new GetApiAccessForEditing($this->getSharedStorage()->get($apiAccessReference)));
+
+        $serializer = CommonFeatureContext::getContainer()->get('serializer');
+        $realData = $serializer->normalize($result);
+
+        DataComparator::assertDataSetsAreIdentical($expectedData, $realData);
```

Comment:
I'm not a fan of this method of comparison because it forces the check to provide all the fields, no partial check is possible which makes the feature scenarios unnecessarily longer in some cases

But it's just a suggestion and a personal opinion, some other behat methods do similar checks so not blocking!